automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala

```diff
@@ -209,6 +233,192 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with AnyFreeSpecLike w
+              "participant",
+              SingleParticipant.entityId,
+              "this",
+              useCallCache = false,
+              deleteIntermediateOutputFiles = false
+            )
+            // clean up: Abort submission
+            register cleanUp Rawls.submissions.abortSubmission(projectName, workspaceName, submissionId)
+
+            // may need to wait for Cromwell to start processing workflows
+            val submissionPatience = PatienceConfig(timeout = scaled(Span(8, Minutes)), interval = scaled(Span(30, Seconds)))
+            implicit val patienceConfig: PatienceConfig = submissionPatience
+
+            // Get workflow ID from submission details
+            val rootWorkflowId: String = eventually {
+              val (status, workflows) = Rawls.submissions.getSubmissionStatus(projectName, workspaceName, submissionId)
+              withClue(s"queue status: ${getQueueStatus()}, submission status: ${getSubmissionResponse(projectName, workspaceName, submissionId)}") {
+                status should (be("Submitted") or be("Done")) // very unlikely it's already done, but let's handle that case.
+                workflows should not be (empty)
+                workflows.head
+              }
+            }
+
+            // Get sub-workflow ids and check the `zones` in workflow options belong to `europe-north1` region
+            val subWorkflowIds: List[String] = eventually {
+              val rootWorkflowMetadata = Rawls.submissions.getWorkflowMetadata(projectName, workspaceName, submissionId, rootWorkflowId)
+              val workflowOptions = parseWorkflowOptionsFromMetadata(rootWorkflowMetadata)
+              val subWorkflowIds = parseSubWorkflowIdsFromMetadata(rootWorkflowMetadata)
+
+              withClue(getWorkflowResponse(projectName, workspaceName, submissionId, rootWorkflowId)) {
+                workflowOptions should include (europeNorth1ZonesPrefix)
+
+                subWorkflowIds should not be (empty)
+                subWorkflowIds.length shouldBe(3)
+
+                subWorkflowIds
+              }
+            }
+
+            // once the sub-workflows have finished running, check
+            //    - the zones for each job that were determined by Cromwell and
+            //    - the worker assigned for the tasks
+            // belong to `europe-north1`
+            eventually {
+              val subWorkflowsMetadata = subWorkflowIds map { Rawls.submissions.getWorkflowMetadata(projectName, workspaceName, submissionId, _) }
+              val subWorkflowCallMetadata = subWorkflowsMetadata flatMap parseCallsFromMetadata
+              val callZones = subWorkflowCallMetadata map { parseRuntimeAttributeKeyFromCallMetadata(_, "zones") }
+              val workerAssignedExecEvents = subWorkflowCallMetadata flatMap { parseWorkerAssignedExecEventsFromCallMetadata }
+
+              subWorkflowsMetadata foreach { x => x should include (""""executionStatus":"Done"""") }
```

Comment:
TOL: if you move everything other than the "is done" check to happen _after_ the `eventually`, the flow would be:

* Wait for completion
* Check once to make sure everything looks correct.

Right now the flow is:

* Wait for everything to look correct.

It's a small difference but it means you'd hopefully get a failure result faster if something goes wrong

Reply:
That's a good point. I will refactor it 👍